ruddock/modules/account/helpers.py

```diff
@@ -99,7 +99,8 @@ def handle_request_account(uid, last_name):
     WHERE uid = :uid
     """)
   result = flask.g.db.execute(query, uid=uid).first()
-  if result is None or result["last_name"].lower() != last_name.lower():
+  name_match = result["last_name"].decode('utf-8').lower() == last_name.lower()
```

Comment:
I expect you'll get an exception here if result is None (happens if the uid is not found), since you've moved this above the None check.